tools/port_scanner.py

```
@@ -28,9 +29,18 @@ def port_scanner():
         print(f"Error: {e}")
 
 
-def main():
-    port_scanner()
+def check_platform_compatibility():
+    supported = True
+    warnings = []
+    try:
+        try:
+            import python_nmap as nmap  # type: ignore
+        except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)